plan/logical_plan_builder.go

```diff
@@ -701,9 +701,17 @@ func (b *planBuilder) buildUnion(union *ast.UnionStmt) LogicalPlan {
 
 	if union.OrderBy != nil {
 		unionPlan = b.buildSort(unionPlan, union.OrderBy.Items, nil)
+		if b.err != nil {
+			b.err = errors.Trace(b.err)
+			return nil
+		}
```

Comment:
I have an idea how about extract this part to this style.

```
package main

import (
	"github.com/juju/errors"
	"fmt"
)

type builder struct {
	err error
}

func (b *builder) b1() {
	b.b2()
	b.traceSelfCallStack()
	return
}

func (b *builder) b2() {
	b.err = errors.New("111")
	return
}

func (b *builder) traceSelfCallStack() {
	if b.err == nil {
		return
	}
	err := errors.NewErrWithCause(b.err, "")
	err.SetLocation(2)
	b.err = &err
	return
}

```
and still, keep every builder self-call has a `b.traceSelfCallStack()` followed..(we have no generic, so we can not `b.selfCall(b.b2)()`)